src/containers/table/Table.js

```diff
@@ -44,6 +44,62 @@ const filterCompaniesWithTags = (companies, tags) => {
     }
 };
 
+
+ const filterCompaniesWithStages = (companies, stages) => {
+    if (stages.length <= 0) {
+        return companies;
+    } else {
+        let comps = [];
+        stages.forEach((stage) => {
+            companies.forEach((comp) => {
+                if (stages.includes(comp.stage)) {
+                    comps.push(comp);
+                }
+            });
+        });
+        return R.uniq(comps);
+    }
+};
+*/
+
+const filterCompanies= (companies, stages, tags) => {
+    if (stages.length <= 0 && tags.length <= 0) {
+        return companies;
+    }else if(stages.length <= 0 && !(tags.length <= 0)) {
+        let comps = [];
+        tags.forEach((tag) => {
+            companies.forEach((comp) => {
+                if (comp.tags && comp.tags.includes(tag)) {
+                    comps.push(comp);
+                }
+            });
+        });
+        return R.uniq(comps);
+    }else if(!(stages.length <= 0) && tags.length <= 0) {
+        let comps = [];
+        stages.forEach((stage) => {
+            companies.forEach((comp) => {
+                if (stages.includes(comp.stageName)) {
```

Comment:
should check if company has stageName at all, otherwise if it encounters undefined it will throw exeption. check previous else if.